server/app.js

```diff
@@ -16,6 +17,15 @@ app.use("/api", api);
 app.use("/auth", auth);
 app.use("/user", passport.authenticate("jwt", { session: false }), user);
 
+// In development environemnt, we use the create-react-app dev server
+// In production, the static build is served from here
+if (process.env.NODE_ENV !== "development") {
+  app.use("/", express.static(path.resolve(__dirname, "../client/build")));
+  app.get("*", (req, res) => {
+    res.sendFile(path.resolve(__dirname, "../client/build", "index.html"));
+  });
```

Comment:
not 100% I understand this - by using `app.get("*")` does that mean **all** requests in our app will be handled by the react router within `index.html`? That's the main reason we added that right? (based on what I read [here](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#serving-apps-with-client-side-routing))

What if we need to call APIs or server some image files? I assume those work with this setup but how?

Reply:
They are handled first. This middleware is only called if a middleware further up the chain doesn't end the request.